physics/sfc_drv_ruc.F90

```diff
@@ -252,117 +249,178 @@ end subroutine lsm_ruc_finalize
-     &       con_cp, con_rv, con_rd, con_g, con_pi, con_hvap, con_fvirt,& !  constants
-     &       weasd, snwdph, tskin, tskin_wat,                           & !  in/outs
-     &       rainnc, rainc, ice, snow, graupel,                         & ! in
-     &       srflag, smois, tslb, sh2o, keepfr, smfrkeep,               & ! in/outs, on RUC levels
-     &       canopy, trans, tsurf, tsnow, zorl,                         &
-     &       sfcqc, sfcdew, tice, sfcqv,                                &
-     &       sncovr1, qsurf, gflux, drain, evap, hflx,                  & ! outputs
-     &       rhosnf, runof, runoff, srunoff,                            &
-     &       chh, cmm, evbs, evcw, sbsno, stm, wetness,                 &
-     &       acsnow, snowfallac,                                        &
-     &       flag_init, flag_restart, errmsg, errflg                    &
+     &       srflag, snoalb, isot, ivegsrc, fice, smcwlt2, smcref2,     &
+     ! --- constants
+     &       con_cp, con_rd, con_rv, con_g, con_pi, con_hvap,           &
+     &       con_fvirt,                                                 &
+     ! for water
+     &       ch_wat, tskin_wat,                                         &
+     ! --- in/outs for ice and land
+     &       semis_lnd, semis_ice,                                      &
+     &       sncovr1_lnd, weasd_lnd, snwdph_lnd, tskin_lnd, sfalb_lnd,  &
+     &       sncovr1_ice, weasd_ice, snwdph_ice, tskin_ice, sfalb_ice,  &
+     ! for land
+     &       smois, tsice, tslb, sh2o, keepfr, smfrkeep,                & ! on RUC levels
+     &       canopy, trans, tsurf_lnd, tsnow_lnd, z0rl_lnd,             &
+     &       sfcqc_lnd, sfcdew_lnd, sfcqv_lnd,                          &
+     &       qsurf_lnd, gflux_lnd, evap_lnd, hflx_lnd,                  &
+     &       runof, runoff, srunoff, drain,                             &
+     &       cm_lnd, ch_lnd, evbs, evcw, stm, wetness,                  &
+     &       snowfallac_lnd,                                            &
+     ! for ice
+     &       sfcqc_ice, sfcdew_ice, sfcqv_ice,                          &
+     &       tice, tsurf_ice, tsnow_ice, z0rl_ice,                      &
+     &       qsurf_ice, gflux_ice, evap_ice, ep1d_ice, hflx_ice,        &
+     &       cm_ice, ch_ice, snowfallac_ice,                            &
+     ! --- out
+     &       acsnow, rhosnf, sbsno,                                     & 
+     &       cmm_lnd, chh_lnd, cmm_ice, chh_ice,                        &
+     !
+     &       flag_iter, flag_guess, flag_init, flag_restart,            &
+     &       flag_cice, frac_grid, errmsg, errflg                       &
      &     )
 
       implicit none
 
 !  ---  constant parameters:
       real(kind=kind_phys), parameter :: rhoh2o  = 1000.0
       real(kind=kind_phys), parameter :: stbolt  = 5.670400e-8
+      real(kind=kind_phys), parameter :: cimin   = 0.15  !--- in GFS
```

Comment:
this value should come in via the argument list, there are standard names for minimum lake ice concentration and minimum sea ice concentration. Certainly also for the Stefan Boltzmann constant, and probably also for `rhoh2o` and `con_tice` (although we need to check the exact values).

Reply:
I have made changes in sfc_drv_ruc.F90 to pass rhowater, con_tice and min_seaice into lsm_ruc_run through the parameter list. Could not find stbolt, looks like many physics subroutines set it locally and to somewhat different values. Should I commit my changes?

Reply:
Yes please! Weird that people use different values of the Boltzmann constant, there is only one that is correct.

Reply:
Commit is done.
I used the value matching the one in physcons: con_sbc    =5.670400e-8_kind_phys.
The module_BL_MYJPBL.F90 uses: STBOLT = 5.67051E-8
Noah LSM uses 5.67e-8
The most exact value is 5.670374419e-8.
What should we do about it?

Reply:
Bang our heads against the next door. Seriously, at least you are using the one from the host model. We should in the future force all physics to use the host model value and correct that one, but not as part of this commit.